### PR TITLE
[AC-7523] changing read replica router to pinning database router

### PR DESCRIPTION
**Pull request by shanbady**
This PR fixes a bug on impact api which shows up in production with a read-replica setup. 

to see the existing issue:
- attempt to get an oauth token from production in rapid succession using the follow command (fill in the values where needed): ```curl --header "Content-Type: application/x-www-form-urlencoded" --header "Accept: application/json; indent=4" --request POST --data "username=<username>&password=<password>&client_id=<client id>&grant_type=password" https://api.masschallenge.org/oauth/token/;```
- you will see that sometimes the command returns a json blob with a token and other times it fails with a 500.

to test the fix:
- I had trouble coming up with a reasonably quick way to test this locally with a read-replica. For the time being I've deployed this branch to the staging api server. running this on staging should not throw a 500 ever: ```curl --header "Content-Type: application/x-www-form-urlencoded" --header "Accept: application/json; indent=4" --request POST --data "username=<username>&password=<password>&client_id=<client id>&grant_type=password" https://stagingapi.masschallenge.org/oauth/token/;```

### CODE REVIEW

**Comment by pbx** [comment]
Reproduced the 500 on prod.  Can't presently cross-check on staging because staging is in use by v1.2.44.

**Comment by pbx** [comment]
@shanbady when you return perhaps deploy this to a test API server rather than staging? That way review won't get sidelined by releases.

**Comment by shanbady** [comment]
@pbx pushing this to a test server requires setting up a read replica database to repro the issue. im going to re-push this to staging unless you think its worth setting it all up on a test machine

**Comment by pbx** [comment]
@shanbady staging seems fine, yeah.

**Comment by shanbady** [comment]
@pbx ive deployed this tag to staging

File: web/impact/impact/settings.py [hunk 1]
```diff
@@ -131,6 +122,20 @@ class Base(Configuration):
         'impact.middleware.MethodOverrideMiddleware',
     ]
 
+    if os.environ.get('READ_REPLICA_DATABASE_URL'):
+        DATABASES.update(values.DatabaseURLValue(  # pragma: no cover
+            alias='read-replica',
+            environ_name='READ_REPLICA_DATABASE_URL').value)
+
+        SLAVE_DATABASES = ['read-replica']  # pragma: no cover
```
Comment (pbx) [inline web/impact/impact/settings.py:130]: FWIW I'd prefer we used "primary/replica" terminology in our codebase instead of "master/slave".

**Review by pbx — APPROVED** [review]
Stress-tested with ten consecutive attempts, no errors. 
```
$ for i in {1..10}; do curl --header "Content-Type: application/x-www-form-urlencoded" --header "Accept: application/json; indent=4" --request POST --data "username=paul%2bsuperuser@masschallenge.org&password=$(pass accelerate-prod/paul+superuser@accelerate.masschallenge.org)&client_id=fU6pHL73OicgutkBNDEe7r9OfbbF3wUjg3EBNDEe7r9OfbbF3wUjg3E63lEV&grant_type=password" https://stagingapi.masschallen done;/oauth/token/;

{"access_token": "drKwPNzSHRWDjJOkRd9HujbZVgxCoq", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "3WHeByNmHPRXPHnUxUBU2kH4YkCowR"}
{"access_token": "uElX0oelZBgh6pN0nGk3VhHyX5RkTf", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "OyybV72WmBhNhf2lO8bkpQDSZkHmce"}
{"access_token": "PD8xucbahuVtVWFZU0yWRE3yPbAbgs", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "8F9yK9F3lHSN4nmLMUGJvrFUGPBOUK"}
{"access_token": "NeaQUqdRiEKD8kdaqPSG3k6ruJhh0B", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "0cQtV0ezPt08YnDV52wYaRsrH1itZR"}
{"access_token": "sn9yfsgQWUGoMi1WFEtsbAukT5xzmC", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "hk6hsxAP9pkY9l3vuhkh2o2ZrWtfUc"}
{"access_token": "lN0ymHaCeDXJ5vRo5BaooR0TWjwuOT", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "zaMhNNQdejS6B5uWd6jyqZuMh06tdg"}
{"access_token": "vhGTXDn189UZRnNDpGGkhjC8gCSbml", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "r6O6Nf0en49qSbTWMLQzhSKvDNoLaJ"}
{"access_token": "S42qgw56OAEN0T3O5feT5hDi3ptCTk", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "TLrpg9MLAoysdrK0WJf6jssu92dh0J"}
{"access_token": "SNGMZqM2NBpuyNV0bF2nojsHlGmSAL", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "sWqpcmuWdbu0TrP4l7Us6iMgrK7lyC"}
{"access_token": "T13UEPRF66R2VQZaC1J1HwNqaqRBJH", "expires_in": 36000, "token_type": "Bearer", "scope": "read write groups", "refresh_token": "Uj1sZS3vHaBLshjtlhngMdx2bcck2q"}
```

**Review by geofrocker — APPROVED** [review]
QA looks good
On `https://api.masschallenge.org`
- Successfully did the first request and it returned the access token payload
- The second request failed with `<h1>Server Error (500)</h1>%`

On `http://stagingapi.masschallenge.org/`
- I did five requests and all of them returned the access token payload

**Pull request merged**